File: Code/Factorial.java
import java.util.*;

public class Factorial {
    public static int fact(int n) {
        // int ans = 1;
        // for (int i = 1; i <= n; i++) {
        //     ans *= i;
        // }
        // return ans;
        if(n>=1) return n*fact(n-1);
        else return 1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter n: ");
        int n = sc.nextInt();
        int ans = fact(n);
        System.out.println("Factorial of " + n + " is " + ans);
        sc.close();
    }
}

File: practice/factorial.cpp
#include <bits/stdc++.h>
using namespace std;

int fact(int n)
{
    // int ans = 1;
    // for (int i = 1; i <= n; i++)
    // {
    //     ans = ans * i;
    // }
    // return ans;
    if (n == 1)
        return 1;
    return n * fact(n - 1);
}

int main()
{
    int n = 5;
    cout << "factorial(" << n << "): " << fact(n) << endl;
    return 0;
}